controller/finalizer.go

```diff
@@ -200,28 +218,6 @@ func createAddFinalizerPatch(obj interface{}, operatorName string) (patch []patc
 	return patch, true, nil
 }
 
-func createRemoveFinalizerPatch(obj interface{}, operatorName string) ([]patchSpec, error) {
-	accessor, err := meta.Accessor(obj)
-	if err != nil {
-		return nil, microerror.Mask(err)
-	}
-
-	finalizerName := getFinalizerName(operatorName)
-	if !containsFinalizer(accessor.GetFinalizers(), finalizerName) {
-		return nil, nil
-	}
```

Comment:
This condition should actually never be true because it is already checked before executing this very function. Then there was not that much code left and I moved it up.